tests/epyccel/test_epyccel_transpose.py

```diff
@@ -166,3 +166,16 @@ def f2(x : 'int[:,:,:]'):
 
     f2_epyc = epyccel(f2, language=language)
     assert f2( x2 ) == f2_epyc( x2 )
+
+def test_transpose_to_inner_indexes(language):
+    def f1(x : 'int[:,:]'):
+        from numpy import transpose, empty
+        n,m = x.shape
+        y = empty((1,m,n,1))
+        y[0,:,:,0] = transpose(x)
+        return y[0,0,0,0], y[0,1,0,0], y[0,0,1,0], y[0,-1,0,0], y[0,0,-1,0]
```

Comment:
I suggest that we pass the array `y` as argument to `f1`, and then we check all components of it with `np.array_equal()`. This makes it easier to write other tests (see below).

Can you please add a few tests to verify other combinations of indices? E.g.
```
y[:, :, 0, 0]
y[:, 1, 0, :]
y[1, :, 0, :]
```
And maybe a 3D slice of a 5D array...

Reply:
Fixed 6da07428

Documentation has also been added